nixos/modules/services/networking/wstunnel.nix

```
@@ -316,13 +316,19 @@ let
             server \
             ${
               lib.cli.toGNUCommandLineShell { } (
-                lib.recursiveUpdate {
-                  restrict-to = map hostPortToString restrictTo;
-                  tls-certificate =
-                    if useACMEHost != null then "${certConfig.directory}/fullchain.pem" else "${tlsCertificate}";
-                  tls-private-key = if useACMEHost != null then "${certConfig.directory}/key.pem" else "${tlsKey}";
-                  websocket-ping-frequency-sec = websocketPingInterval;
-                } extraArgs
+                lib.recursiveUpdate
+                  {
+                    restrict-to = map hostPortToString restrictTo;
+                    websocket-ping-frequency-sec = websocketPingInterval;
+                  }
+                  (
+                    lib.optionalAttrs (enableHTTPS) {
+                      tls-certificate =
+                        if useACMEHost != null then "${certConfig.directory}/fullchain.pem" else "${tlsCertificate}";
+                      tls-private-key = if useACMEHost != null then "${certConfig.directory}/key.pem" else "${tlsKey}";
+                    }
+                    // extraArgs
+                  )
```

I think this would partially break the effect of `lib.recursiveUpdate`. How about something like this instead?

```suggestion
                lib.recursiveUpdate
                  {
                    restrict-to = map hostPortToString restrictTo;
                    websocket-ping-frequency-sec = websocketPingInterval;
                    tls-certificate = if !enableHTTPS then null
                                 else if useACMEHost != null then "${certConfig.directory}/
                                 else "${tlsCertificate}";
                   tls-private-key = if !enableHTTPS then null
                                else if useACMEHost != null then "${certConfig.directory}/key.pem"
                                else "${tlsKey}";
                  } extraArgs;
```

Alternatively you could also use `//` between the two static attrsets, and `recursiveUpdate` the result with `extraArgs`.

---

Nice, I like this suggestion. Will implement and re-tag you!

---

Used your first suggestion while appeasing the Nix formatter.